src/lib/validations/is-multisite-domain-mapped.ts

```diff
@@ -115,18 +114,18 @@ export async function isMultisitePrimaryDomainMapped(
 			error_type: 'GraphQL-MappedDomain-Check-failed',
 			gql_err: GraphQlError,
 		} );
+		// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
 		exit.withError( `StartImport call failed: ${ GraphQlError }` );
 	}
-	if ( ! Array.isArray( res?.data?.app?.environments ) ) {
+	if ( ! Array.isArray( res.data.app?.environments ) ) {
 		return false;
 	}
 
-	const environments = res.data.app.environments;
+	const environments = res.data.app!.environments;
```

Comment:
nit: I'd usually do `res.data.app?.environments || []` rather than `!` to fix the type error here, though I guess this is more of a parker bug - because the `app` query never returns null and should've been of type `App!` in the schema.

I'm just paranoid 😁 but I guess I'd rather have zero chance of the undefined access from happening?

Reply:
We have a check a couple lines above:

```ts

if ( ! Array.isArray( res.data.app?.environments ) ) {
    return false;
}
```

If `res.data.app` is undefined, we don't reach this line. That's why `!`.